sycl/include/sycl/types.hpp

```diff
@@ -1488,6 +1488,15 @@ class SwizzleOp {
       SwizzleOp<const VecT, GetOp<DataT>, GetOp<DataT>, GetOp, Indices...>;
 
 public:
+  using element_type = DataT;
+
+  const DataT &operator[](int i) const { return (*m_Vector)[i]; }
+
+  template <typename _T = VecT>
+  std::enable_if_t<!std::is_const_v<_T>, DataT> &operator[](int i) {
+    return (*m_Vector)[i];
```

Comment:
I think `SwizzleOp::getValue` should be used to evaluate the swizzled operation instead of indexed `m_Vector` here. At least for how I interpret swizzle should work, the following test cases would not work: 
```c++
  sycl::int2 x = {1, 2};
  assert((x.swizzle<0,1>() + x)[0] == 2)
  assert(x.swizzle<1,0>()[0] == 2)